### DISCUSSION TITLE
Smoother Zoom

### DISCUSSION THREAD

**Discussion by Kalnon26** [sumatrapdfreader/sumatrapdf#3598] [General]
Hello, I just starting using Samatra and I really enjoy it because it's so light and easy to use.  I'm getting heavily into flight simulation and that requires constant reading of manuals of all kinds, I have a massive collection of pdf files and I like being able to have many open at once and the app still be light.  My only complaint is that I wish the zoom feature worked more smoothly.  When I zoom in and out, it is in large increments.  It would help me a lot and it would be wonderful to have smooth zoom or at least make it change in smaller increments.

Thank you to the developer for you great little pdf reader.

**Comment by GitHubRulesOK** [comment]
PDF rendering is naturally based on scale factors so when reading interlaced images it looks rough or rewriting the screen full of data for each small zoom takes time so to be fast a viewer needs to effectively chunk from one selection to next selection.

However there is an option to lessen the stepping in advanced settings so start test with "ZoomIncrement = 10" (otherwise at 1 it will be frustratingly slow) then reduce towards 1 as much as you need for "smoother"

**Reply by Kalnon26** [reply]
Thank you so much for the help, I greatly appeciate it.   Setting ZoomIncrement = 10 is perfect for me.  The increment stepping is plenty good enough at that setting and it doesn't seem to slow things down at all as far as I can tell.  The default value of 0 was just way to much for me personally., just one step and it would be either much too big or much too small.

**Reply by GitHubRulesOK** [reply]
since using Increments are now % so not uniform, if you need to force the zoom level to a setting like 50% or 200% you can use Ctrl+Y to set an entry or the zoom menu

**Reply by Kalnon26** [reply]
That's a good feature, I didn't know about that, thanks.  I will be using that and I'll probably be messing around with the Zoomincrement setting some more just out of curiosity.  I like messing around with things like this, I just never realized it was possible with this app.

**Reply by tactilis** [reply]
Be aware that SumatraPDF overwrites the advanced options file _SumatraPDF-settings.txt_ every time you exit the app, so take care to coordinate your editing of that file and the running of SumatraPDF.

**Reply by GitHubRulesOK** [reply]
@tactilis 
any edit made to settings "generally" should be seen on save (there are a few exceptions needing close re-open) but more often than not any edits and Ctrl+S to settings are preserved and only later session data is overwritten

**Reply by Kalnon26** [reply]
Ok I understand, thank you for the advice.

**Comment by Kalnon26** [comment]
I have another question about Sumatra, is it possible to set the page view to be able to see two pages next to each other side by side?

I see there are a couple options to adjust page width that I'm seeing but not side by side.  Some of my aircraft manuals are made to be viewed this way when they show things to study like pictorial diagrams and views of an instrument panel (which are very wide), they have to seperate it into two pages side by side.

**Reply by GitHubRulesOK** [reply]
Two options available different ways (you can also set as default on opening) so Facing or Book (page 1 same as a cover, is held separate to 2&3) to reverse from left to right for arabic/hebrew etc is oddly Toggle Manga Mode as its legacy from comic books !

Advanced setting for `PageSpacing = 4 4` can close the gap to 0 (page gaps in small numbers of X Y pixels, beware a silly number like negative or high positive is known to cause needing new setting file!)

**Reply by Kalnon26** [reply]
Lol from comic books, I had no idea.  I'm so glad that there is a way.  Thank you so much for all of these tips and advice, I greatly appreciate your help.

**Reply by GitHubRulesOK** [reply]
varies with version but for flipping left with right I use shortcuts

for illustration in action see https://github.com/sumatrapdfreader/sumatrapdf/discussions/3298 also https://github.com/sumatrapdfreader/sumatrapdf/discussions/2843 and my shortcut is here
https://github.com/sumatrapdfreader/sumatrapdf/discussions/2775#discussioncomment-3066563

**Reply by Kalnon26** [reply]
That is extremely helpful, thank you so much.  I never noticed the three horizontal bars.  Unlike the pictures in your links, mine is colored black, hidden in the black bar at the top and it's hidden unless I hover over it with the mouse cursor.  So many options I never knew about.
